### DISCUSSION TITLE
Provide more help and testing feature for e-mail setup

### DISCUSSION THREAD

**Discussion by jaanus** [mastodon/mastodon#23345] [General]
I have set up several Mastodon instances by now. One of the trickier parts is SMTP setup. Lots of things can go wrong there, and you don’t really know if things are working correctly. The default installation script offers to send one test e-mail after entering SMTP configuration.

I have some ideas for how to improve the admin user experience. Before writing them up as issues, I thought I’d just drop them here.

## Offer a way to send a test e-mail to yourself

I’d like to invoke a test e-mail to some address that just says “hey, this works”. Similarly to the test e-mail that is sent during setup, but at any later time. This could be with `tootctl` or admin web UI or any other place I can access as admin, doesn’t really matter.

## Easy way to see e-mail log

I am sure the e-mail log is available somewhere in the system. I found related open issue which discusses it: https://github.com/mastodon/mastodon/issues/7323

As an admin, I’d like to filter the logs for successful and more importantly failed e-mail send events, with some easily copypastable commands.

## Cover the above two in docs

The docs at https://docs.joinmastodon.org/ don’t have much about e-mail. Docs could cover the above two items.

**Comment by fweng322** [comment]
I'm still struggling with this... the SMTP settings worked okay but suddenly nothing sent anymore, and I couldn't find out why.  I tried to fix the SMTP settings but couldn't have any way to test if it works or not.  Really frustrating, actually.

**Comment by bheerssen** [comment]
Has there been any movement on this?

**Comment by thommierother** [comment]
I started with Mastodon 4.5.x and upgraded to 4.6.3 recently. Unfortunately, the smtp configuration for mastodon and especially the debugging options for smtp are still a big mess. There is only rudimentary documentation for SMTP setup, no options to send test messages and no debugging help if mail transport fails. Other rails applications (see Discourse at https://github.com/discourse/discourse) are much easier to debug for smtp transport issues. E-Mail communication is essential. If it fails during the setup of a new instance, it seriously prevents the further spread of self-hosted mastodon instances.

If one takes into account that mastodon gets some funding from official sources (e.g. from @SovTechFun), I am really surprised that this hasn't been sorted out yet.
